spark/v3.2/spark/src/test/java/org/apache/iceberg/spark/source/TestSparkCatalogCacheExpirationEnabled.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.spark.source;
+
+import java.util.Map;
+import org.apache.iceberg.CachingCatalog;
+import org.apache.iceberg.CatalogProperties;
+import org.apache.iceberg.catalog.Catalog;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.spark.SparkCatalog;
+import org.apache.iceberg.spark.SparkCatalogTestBase;
+import org.apache.spark.sql.connector.catalog.TableCatalog;
+import org.assertj.core.api.Assertions;
+import org.junit.Test;
+import org.junit.runners.Parameterized;
+
+public class TestSparkCatalogCacheExpirationEnabled extends SparkCatalogTestBase {
```

Comment:
Here as well, I don't think that we need `SparkCatalogTestBase` Instead, can you use `SparkTestBaseWithCatalog`?

Reply:
Actually, it looks like you refactored the tests and you're trying to use the initial catalog config passed to create the test suite as parameters to be tested. That's a _really_ expensive way to run these tests because it starts up a lot of services, which is why I wanted to use `SparkTestBaseWithCatalog` instead -- that only starts Spark and everything once.

Here, you should create a new catalog in each test method with different settings and a different name, then test that it functions as you expect. These test classes should all be cases of a single class.

Reply:
Yeah. The problem I had with `SparkTestBaseWithCatalog` was that it didn't allow for overriding the catalog configuration. At least not in the ways that I tried it.

I'll take another look with that in mind.

Reply:
Yeah, that's just not the right way to test catalog properties. Instead, create a new catalog and test it, within the same Spark session.

Reply:
Oh good point. I've been sick the past few days and hadn't thought of that. I can also test some of these within `CachingCatalog`'s test suite too.

Reply:
Please take a look when you can @rdblue. I've moved all of the Spark based tests into one file, as mentioned.

I used a BeforeClass function to set up the other catalogs.